packages/language-server/test/completion-items-spec.ts

```diff
@@ -151,41 +487,149 @@ describe("the UI5 language assistant Code Completion Services", () => {
                           xmlns="sap.m"> 
                           <List> <te⇶`;
     const suggestions = getSuggestions(xmlSnippet);
-    const suggestionNames = map(suggestions, suggestion => suggestion.label);
+    const suggestionNames = map(suggestions, suggestion => ({
+      label: suggestion.label,
+      replacedText: getTextInRange(xmlSnippet, suggestion.textEdit?.range)
+    }));
+    const suggestionKinds = uniq(
+      map(suggestions, suggestion => suggestion.kind)
+    );
+
+    expect(suggestionNames).to.deep.equalInAnyOrder([
+      { label: "contextMenu", replacedText: "te" },
+      { label: "items", replacedText: "te" },
+      { label: "swipeContent", replacedText: "te" }
+    ]);
+
+    expect(suggestionKinds).to.deep.equal([CompletionItemKind.Text]);
+  });
+
+  it("will get completion values for UI5 aggregation when the cursor is in the middle of a name", () => {
+    const xmlSnippet = `<mvc:View 
+                          xmlns:mvc="sap.ui.core.mvc" 
+                          xmlns="sap.m"> 
+                          <List> <te⇶Menu`;
+    const suggestions = getSuggestions(xmlSnippet);
+    const suggestionNames = map(suggestions, suggestion => ({
+      label: suggestion.label,
+      replacedText: getTextInRange(xmlSnippet, suggestion.textEdit?.range)
+    }));
     const suggestionKinds = uniq(
       map(suggestions, suggestion => suggestion.kind)
     );
 
     expect(suggestionNames).to.deep.equalInAnyOrder([
-      "contextMenu",
-      "items",
-      "swipeContent"
+      { label: "contextMenu", replacedText: "teMenu" },
+      { label: "items", replacedText: "teMenu" },
+      { label: "swipeContent", replacedText: "teMenu" }
     ]);
 
     expect(suggestionKinds).to.deep.equal([CompletionItemKind.Text]);
   });
 
   it("will get completion values for UI5 xmlns key namespace", () => {
+    function getFQNFromDetail(detail: string | undefined): string | undefined {
+      if (detail === undefined) {
+        return detail;
+      }
+      return detail.replace("(deprecated) ", "").replace("(experimental) ", "");
+    }
+
     const xmlSnippet = `<mvc:View 
                           xmlns:mvc="sap.ui.core.mvc" 
                           xmlns:u⇶`;
     const suggestions = getSuggestions(xmlSnippet);
-    const suggestionNames = map(suggestions, suggestion => suggestion.label);
+    const suggestionNames = map(suggestions, suggestion => ({
+      label: suggestion.label,
+      namespaceName: getFQNFromDetail(suggestion.detail),
+      replacedText: getTextInRange(xmlSnippet, suggestion.textEdit?.range)
+    }));
     const suggestionKinds = uniq(
       map(suggestions, suggestion => suggestion.kind)
     );
 
     expect(suggestionNames).to.deep.equalInAnyOrder([
-      "unified",
-      "upload",
-      "util",
-      "ux3",
-      "uxap",
-      "ubc",
-      "ui",
-      "ui", // This "ui" is in a different parent
-      "ui5",
-      "ulc"
+      {
```

Comment:
this is a good example of a test I would like to know the textEdit in
to understand what is actually inserted once the user selects a suggestion

Reply:
I'll add it here

Reply:
Done